akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala

```diff
@@ -221,10 +221,18 @@ abstract class EventSourcedBehavior[Command, Event, State] private[akka] (
       if (handler.isEmpty) behavior
       else behavior.receiveSignal(handler.handler)
 
-    if (onPersistFailure.isPresent)
-      behaviorWithSignalHandler.onPersistFailure(onPersistFailure.get)
-    else
-      behaviorWithSignalHandler
+    val withSignalHandler =
+      if (onPersistFailure.isPresent)
+        behaviorWithSignalHandler.onPersistFailure(onPersistFailure.get)
+      else
+        behaviorWithSignalHandler
+
+    if (stashCapacity.isPresent()) {
```

Comment:
```suggestion
    if (stashCapacity.isPresent) {
```    


I tried to use `map/orElse` for this, but surprisingly type inference is not working as expected. 🤷

Reply:
This is not an important suggestion. Can skip if you prefer. 

I added here because intellij gave the suggestion. I'm just passing it over. 😄